tokio/src/sync/watch.rs

```diff
@@ -90,9 +90,9 @@ pub struct Sender<T> {
 /// Returns a reference to the inner value.
 ///
 /// Outstanding borrows hold a read lock on the inner value. This means that
-/// long lived borrows could cause the produce half to block. It is recommended
+/// long lived borrows could cause the producer half to block. It is recommended
 /// to keep the borrow as short lived as possible. Additionally, if you are
-/// running in an environment that allows `!Send` futures, you must ensure that
+/// running in an environment that disallows `!Send` futures, you must ensure that
 /// the returned `Ref` type is never held alive across an `.await` point.
```

Comment:
This is correct as written.

Reply:
But if the environment allows `!Send` futures, then there is no restriction for holding a reference to `Ref` across an `.await` point.

Reply:
Holding a `Ref` across an `.await` is _always_ bad because it can lead to a deadlock, but if `!Send` futures are disallowed, then you can't do it by accident since the compiler will catch it.

Reply:
Aha, I see this may lead to a panic or a deadlock... That's, I was missing that detail. Maybe I should extend this comment to make that more clear what the aftermath is? WDYT?

Reply:
That's fine. (It can only lead to a deadlock, not a panic.)

Reply:
Added a small clarification about the deadlock, and added a dash to `long-lived` and `short-lived` words as per the grammar.

I've noticed that this paragraph that warns about holding a ref across an `.await` point is repeated in several places. Some of them slightly differed from the doc comment on the `struct Ref`. I've updated them all to the same wording for consistency ✔️

Reply:
@Darksonn waiting for your review